mod/consensus/pkg/cometbft/service/validate_deposits.go

```
@@ -0,0 +1,78 @@
+func validateDeposits(deposits []types.Deposit) error {
+	if len(deposits) == 0 {
+		return errors.New("at least one deposit is required")
+	}
+
+	seenPubkeys := make(map[string]bool)
+
+	for i, deposit := range deposits {
+		depositIndex := deposit.GetIndex()
+		//#nosec:G701 // realistically fine in practice.
+		// Validate index matches position
+		if depositIndex.Unwrap() != uint64(i) {
```

golangci-lint (1.62.2)

[high] G115: integer overflow conversion int -> uint64

(gosec)